src/main/java/org/akaza/openclinica/service/ImportServiceImpl.java

```diff
@@ -1432,47 +1447,95 @@ private Object validateItem(ImportItemDataBean itemDataBean, CrfBean crf, EventC
+                if(parentDn == null) {
+                    int childNotesCount = discrepancyNoteBean.getChildNotes().size();
+                    if(childNotesCount > 0){
+                        queryBean.setComment(discrepancyNoteBean.getChildNotes().get(childNotesCount - 1).getDetailedNote());
+                        UserAccount userAccount1 = userAccountDao.findByUserName(discrepancyNoteBean.getChildNotes().get(childNotesCount - 1).getOwnerUserName());
+                        container.setUser(userAccount1);
+                    }
+                    parentDn = (DiscrepancyNote) queryService.createQuery(helperBean, queryBean, true);
+//                    parentDn.setThreadNumber(discrepancyNoteBean.getThreadNumber());
+                    parentDn.setDisplayId(discrepancyNoteBean.getDisplayId());
+                    parentDn = discrepancyNoteDao.saveOrUpdate(parentDn);
+                    parentDn.setThreadUuid(UUID.randomUUID().toString());
+                    helperBean.setDn(parentDn);
+                    queryService.saveQueryItemDatamap(helperBean);
+                }
+                String lastComment ="";
+                UserAccount lastuser = null;
+                for(ChildNoteBean childNoteBean  : discrepancyNoteBean.getChildNotes()){
+                    DiscrepancyNote childDN = null;
+                    if(childNoteBean.getDisplayId()!= null )
+                        childDN = discrepancyNoteDao.findByDisplayId(childNoteBean.getDisplayId());
+                    if(childDN != null)
+                        continue;
+                    lastuser = userAccountDao.findByUserName(childNoteBean.getOwnerUserName());
+                    queryBean.setComment(childNoteBean.getDetailedNote());
+                    queryBean.setStatus(childNoteBean.getStatus());
+                    container.setUser(lastuser);
+                    childDN = queryService.createQuery(helperBean, queryBean, false);
+                    if(childNoteBean.getUserRef() != null) {
+                        UserAccount assignedUser = userAccountDao.findByUserName(childNoteBean.getUserRef().getUserName());
+                        childDN.setUserAccount(assignedUser);
+                    }
+                    childDN.setParentDiscrepancyNote(parentDn);
+                    childDN.setDisplayId(childNoteBean.getDisplayId());
+                    childDN.setThreadUuid(parentDn.getThreadUuid());
+                    childDN = discrepancyNoteDao.saveOrUpdate(childDN);
+                    lastComment = childDN.getDetailedNotes();
+
+                    // update Item data map
+                    helperBean.setDn(childDN);
+                    helperBean.setParentDn(parentDn);
+                    queryService.saveQueryItemDatamap(helperBean);
+                }
+                parentDn.setDetailedNotes(lastComment);
+                parentDn.setUserAccountByOwnerId(lastuser);
+                discrepancyNoteDao.saveOrUpdate(parentDn);
```

Comment:
move the code to create a child note to a new method so it's easier to maintain and potentially you can reuse this code for creating parent note as well?